code/modules/mob/living/simple_animal/hostile/giant_spider.dm

```diff
@@ -524,8 +536,59 @@ s
 		remove_ranged_ability()
 		return TRUE
 
-/obj/effect/proc_holder/wrap/on_lose(mob/living/carbon/user)
-	remove_ranged_ability()
+/obj/effect/proc_holder/spider/throw_web
+	name = "Throw web"
+	panel = "Spider"
+	desc = "Throw a sticky web at potential prey to immobilize them temporarily"
+	ranged_mousepointer = 'icons/effects/throwweb_target.dmi'
+	action_icon = 'icons/mob/actions/actions_animal.dmi'
+	action_icon_state = "throw_web_0"
+	action_background_icon_state = "bg_alien"
+
+/obj/effect/proc_holder/spider/throw_web/activate(mob/living/user)
+	var/message
+	if(active)
+		message = "<span class='notice'>You discard the webbing</span>"
+		remove_ranged_ability(message)
+	else
```

Comment:
early return if(active) would be better

Reply:
Why exactly? 

This forces the spider in a state of web throwing until they throw the web. I don't see many situations in which they would rather discard the web than just lay it down somewhere, but options aren't necessarily bad.

Reply:
I think I meant from a code readability standpoint - I meant to say if(!active) since the else case just has return it looks weird

Reply:
Ah yep. Took that else out entirely now too.

Reply:
Please add back the original active logic